sql/core/src/main/java/org/apache/spark/sql/execution/UnsafeFixedWidthAggregationMap.java

```diff
@@ -225,4 +232,93 @@ public void printPerfMetrics() {
     System.out.println("Total memory consumption (bytes): " + map.getTotalMemoryConsumption());
   }
 
+  /**
+   * Sorts the key, value data in this map in place, and return them as an iterator.
+   *
+   * The only memory that is allocated is the address/prefix array, 16 bytes per record.
+   */
+  public KVIterator<UnsafeRow, UnsafeRow> sortedIterator() {
+    int numElements = map.numElements();
+    final int numKeyFields = groupingKeySchema.size();
+    TaskMemoryManager memoryManager = map.getTaskMemoryManager();
```

Comment:
can you name this `taskMemoryManager` to disambiguate from `ShuffleMemoryManager`?